### ISSUE TITLE
Use NS_ENUM to declare enums

### ISSUE THREAD

**Issue by segiddins** [rentzsch/JREnum#8]
Use `typedef NS_ENUM` instead of the old-fashion `typedef`

**Comment by rentzsch** [comment]
Yup yup. My current thinking is to put this into a semver 2.x since `JREnum` & `JREnumDeclare` would need an additional `ENUM_TYPE` argument.

**Comment by rentzsch** [comment]
While I'm in there I'll probably add a `JROptions` to mirror `NS_OPTIONS`.
